python/pyiceberg/schema.py

```diff
@@ -317,6 +331,97 @@ def primitive(self, primitive: PrimitiveType) -> T:
         """Visit a PrimitiveType"""
 
 
+class SchemaVisitorPerPrimitiveType(SchemaVisitor[T], ABC):
```

Comment:
Seems reasonable to me. Should we have two visitor classes or do we not need to handle individual primitives sometimes?